openbb_terminal/common/prediction_techniques/expo_model.py

```diff
@@ -0,0 +1,131 @@
+""" Probablistic Exponential Smoothing Model"""
+__docformat__ = "numpy"
+
+import logging
+from typing import Any, Tuple, Union, List
+
+import numpy as np
+import pandas as pd
+from darts import TimeSeries
+from darts.models import ExponentialSmoothing
+from darts.dataprocessing.transformers import MissingValuesFiller
+from darts.utils.utils import ModelMode, SeasonalityMode
+from darts.metrics import mape
+
+from openbb_terminal.decorators import log_start_end
+from openbb_terminal.rich_config import console
+
+
+TRENDS = ["N", "A", "M"]
+SEASONS = ["N", "A", "M"]
+PERIODS = [4, 5, 7]
+DAMPED = ["T", "F"]
+
+logger = logging.getLogger(__name__)
+
+
+@log_start_end(log=logger)
+def get_expo_data(
+    data: Union[pd.Series, pd.DataFrame],
+    trend: str = "A",
+    seasonal: str = "A",
+    seasonal_periods: int = None,
+    damped: str = "F",
+    n_predict: int = 30,
+    start_window: float = 0.65,
+    forecast_horizon: int = 3,
+) -> Tuple[List[float], List[float], Any, Any]:
+
+    """Performs Probabalistic Exponential Smoothing forecasting
+    This is a wrapper around statsmodels Holt-Winters' Exponential Smoothing;
+    we refer to this link for the original and more complete documentation of the parameters.
+
+    https://unit8co.github.io/darts/generated_api/darts.models.forecasting.exponential_smoothing.html?highlight=exponential
+
+    Parameters
+    ----------
+    data : Union[pd.Series, np.ndarray]
+        Input data.
+    trend: str
+        Trend component.  One of [N, A, M]
+        Defaults to ADDITIVE.
+    seasonal: str
+        Seasonal component.  One of [N, A, M]
+        Defaults to ADDITIVE.
+    seasonal_periods: int
+        Number of seasonal periods in a year
+        If not set, inferred from frequency of the series.
+    damped: str
+        Dampen the function
+    n_predict: int
+        Number of days to forecast
+    start_window: float 
+        Size of sliding window from start of timeseries and onwards
+    forecast_horizon: int
+        Number of days to forecast when backtesting and retraining historical
+
+    Returns
+    -------
+    List[float]
+        Adjusted Data series
+    List[float]
+        List of predicted values
+    Any
+        Fit Prob. Expo model object.
+    """
+
+    filler = MissingValuesFiller()
+    data["date"] = data.index  # add temp column since we need to use index col for date
+    ticker_series = TimeSeries.from_dataframe(
+        data,
+        time_col="date",
+        value_cols=["AdjClose"],
+        freq="B",
+        fill_missing_dates=True,
```

Comment:
It is fine to leave this here, but it might incur a small performance penalty. If you happen to already have the guarantee beforehand that your DataFrame does not have missing dates (even if it has missing values), you can consider setting this to False.